pkg/server/settingswatcher/overrides.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package settingswatcher
+
+// OverridesMonitor is an interface through which the settings watcher can
+// receive setting overrides. Used for non-system tenants.
+//
+// The expected usage is to listen for a message on NotifyCh(), and use
+// Current() to retrieve the updated list of overrides when a message is
+// received.
+type OverridesMonitor interface {
+	// NotifyCh returns a channel that receives a message any time the current set
+	// of overrides changes.
+	NotifyCh() <-chan struct{}
+
+	// Overrides retrieves the current set of setting overrides, as a map from
+	// setting key to RawValue. Any settings that are present must be set to the
+	// overridden value.
+	Overrides() map[string]RawValue
```

Comment:
just to confirm my understanding: the expected implementation of OverridesMonitor will internally keep tenant-specific overrides and all-tenant overrides and return the merged map here, right?

Reply:
OverridesMonitor will do an RPC through the tenant connector to listen for overrides. The host side of that API will "merge" the overrides. The tenant side won't be able to tell what kind of override it was.

Reply:
Hm. I think we should reconsider that -- but it doesn't really change the API used here either way, so I'm happy to wait on that until the implementation of the monitor and its corresponding host RPC API. 

(basically I just don't want deleting one all-tenants override to force a read of the every tenant's possible per-tenant override for that setting, so I want them to be stateful and keep track of two layers)

Reply:
I agree with that goal. My plan was along the lines of a single process per KV node that watches the entire `tenant_settings` table and keeps all the information in memory, updating any connected tenants as necessary. I expect that we won't have a ton of entries in that table - thinking we'll have a handful of all-tenant settings and a few one-off tenant-specific settings.